client/landing/stepper/declarative-flow/flows/domain/domain.ts

```diff
@@ -165,6 +165,13 @@ const domain: FlowV2< typeof initialize > = {
 						return navigate( destination as typeof currentStepSlug );
 					}
 
+					// Handle skip to plan when user needs paid plan for ownership verification
+					if ( providedDependencies && 'skipToPlan' in providedDependencies ) {
```

Comment:
If they need a plan, shouldn't we add the mapping to the cart anyway?

Reply:
The issue is that a successful verification would create the mapping before checkout, and the user would end up with a connection but without a plan if they skip the checkout